Signal/src/ViewControllers/OWSViewController.h

```diff
@@ -8,6 +8,13 @@ NS_ASSUME_NONNULL_BEGIN
 
 @interface OWSViewController : UIViewController
 
+// We often want to pin one view to the bottom guide
+// of a view controller BUT adjust its location upward
+// if the keyboard appears.
+//
+// Use this method in lieu of autoPinToBottomLayoutGuideOfViewController:
+- (void)autoPinViewToBottomGuideOrKeyboard:(UIView *)view;
```

Comment:
A big change!  We often want to pin the bottom of our views' content to the bottom of the screen, but move it upward when the keyboard is presented.  This method will allow us to easily and consistently do that throughout our views.

We haven't done a good job of this historically; this will fix a bunch of janky animations.  

I'm looking forward to applying this method to more views.